examples/robot_example.cpp

```diff
@@ -45,9 +45,9 @@ class SimpleRobotControlLoop : public kodlab::mjbots::MjbotsControlLoop<ManyMoto
         }
     }
 
-    void ProcessInput() override
+    void ProcessInput(const ModeInput &input_data) override
```

Comment:
I like this cause it forces the user to be explicit and reminded about their type and also has better memory control.  win win in my book, even if it isn't quite as easy to use templatewise.